src/vm/tests/conversions.rs

```diff
@@ -230,3 +232,137 @@ fn test_simple_buff_to_uint_be() {
         .into()
     );
 }
+
+#[test]
+fn test_simple_string_to_int() {
+    let good1_test = r#"(string-to-int "-1")"#;
+    assert_eq!(Value::Int(-1), execute_v2(good1_test).unwrap().unwrap());
+
+    let good1_test = r#"(string-to-int u"-1")"#;
+    assert_eq!(Value::Int(-1), execute_v2(good1_test).unwrap().unwrap());
+
+    let no_args_test = r#"(string-to-int)"#;
+    assert_eq!(
+        execute_v2(no_args_test).unwrap_err(),
+        CheckErrors::IncorrectArgumentCount(1, 0).into()
+    );
+
+    let bad_value_error_test = r#"(string-to-int "a")"#;
+    assert_eq!(
+        execute_v2(bad_value_error_test).unwrap_err(),
+        CheckErrors::InvalidCharactersDetected.into()
+    );
+
+    let wrong_type_error_test = r#"(string-to-int 1)"#;
+    assert_eq!(
+        execute_v2(wrong_type_error_test).unwrap_err(),
+        CheckErrors::UnionTypeValueError(
+            vec![
+                TypeSignature::max_string_ascii(),
+                TypeSignature::max_string_utf8(),
+            ],
+            Value::Int(1)
+        )
+        .into()
+    );
+}
+
+#[test]
+fn test_simple_string_to_uint() {
+    let good1_test = r#"(string-to-uint "1")"#;
+    assert_eq!(Value::UInt(1), execute_v2(good1_test).unwrap().unwrap());
+
+    let good2_test = r#"(string-to-uint u"1")"#;
+    assert_eq!(Value::UInt(1), execute_v2(good2_test).unwrap().unwrap());
+
+    let no_args_test = r#"(string-to-uint)"#;
+    assert_eq!(
+        execute_v2(no_args_test).unwrap_err(),
+        CheckErrors::IncorrectArgumentCount(1, 0).into()
+    );
+
+    let bad_value_error_test = r#"(string-to-uint "a")"#;
+    assert_eq!(
+        execute_v2(bad_value_error_test).unwrap_err(),
+        CheckErrors::InvalidCharactersDetected.into()
+    );
+
+    let wrong_type_error_test = r#"(string-to-uint 1)"#;
+    assert_eq!(
+        execute_v2(wrong_type_error_test).unwrap_err(),
+        CheckErrors::UnionTypeValueError(
+            vec![
+                TypeSignature::max_string_ascii(),
+                TypeSignature::max_string_utf8(),
+            ],
+            Value::Int(1)
+        )
+        .into()
+    );
```

Comment:
Same here -- please check a wider array of bad strings.

Reply:
i added some more

Reply:
Can you add a test for `(string-to-uint -1)` to verify that it fails typecheck?

Reply:
It also might be worth adding a test that tries to convert a too-big string into an integer (or an unsigned integer), and verifies that it succeeds for the maximal size but fails for a string with one byte more.  That way, we'll be covering the boundary condition.